articles/roles-data-science/index.md

```diff
@@ -0,0 +1,125 @@
+
+![big data](/engineering-education/roles-data-science/big_data_tech.jpg)
+
+[Image Source](https://www.karmelsoft.com/skills-every-big-data-architect-needs/)
+
+The image above shows the various technologies a data architect deals with. There is a logical explanation for the entire process of what a data architect does. The steps can be summarized into five key points:
+   1. Data ingestion: For example, consider the millions of likes, comments, and shares that Facebook deals with. Therefore, scalable infrastructure is required to support such massive amounts of input data. Several software frameworks support the efficient ingestion of data.
+   2. Data Storage: Storing petabytes of data efficiently is necessary for easy retrieval. Therefore, many ecosystems have been built around this aspect.
+   3. Data processing: Processing data refers to ensuring models developed by ML scientists and engineers take in the input and give the desired output.
+   4. Data Analysis: Certain algorithms such as recommendation engines analyze the viewing data in real-time and give millions of recommendations every second. Just imagine the scale of operations.
+   5. Data exploration: Many processes are automated to ensure the other three roles have minimal trouble exploring data. For example, categorical features are encoded using various encoding schemes.
+
+Having considered the various responsibilities, let's look at some of the technologies used to implement.
+
+### Tools and Technologies Commonly Used
+1. Machine Learning Engineer Tools:
+   - Exploratory Data Analysis Tools: [Excel](https://www.microsoft.com/en-us/microsoft-365/excel), [Matlpotlib](https://matplotlib.org/), [Seaborn](https://seaborn.pydata.org/), [H20](https://www.h2o.ai/)
+   - Embedded Systems: [Raspberry-pi](https://www.raspberrypi.org/), [Beaglebone Black](https://beagleboard.org/black),[Nvidia Jetson](https://developer.nvidia.com/buy-jetson), [Movidius](https://www.intel.com/content/www/us/en/products/processors/movidius-vpu.html)
+   - Programming environments: [Anaconda](https://www.anaconda.com/), [H20.ai](https://www.h2o.ai/)
+   - Programming Languages: [Python](https://www.python.org/), C/C++
+   - Commonly used libraries: [Tensorflow](https://www.tensorflow.org/), [PyTorch](https://pytorch.org/), [Tensorflow Lite](https://www.tensorflow.org/lite), [Scikit-learn](https://scikit-learn.org/)
+2. Machine Learning Scientist Tools:
+   - Machine Learning Frameworks: [Tensorflow](https://www.tensorflow.org/), [PyTorch](https://pytorch.org/), [MxNet](https://mxnet.apache.org/), [Caffe](http://caffe.berkeleyvision.org/)
+   - Parallel and Distributed Programming Frameworks: [CUDA](https://developer.nvidia.com/cuda-downloads), [OpenGL](https://www.opengl.org/)
+3. Data Analysts Tools:
+   - Exploratory Data Analysis:
+   - Databases Management Systems:
```

Comment:
I think you may have forgot to include any 'Exploratory Data Analysis' tools? @lalith1403

Reply:
I had mentioned the same under machine learning engineer tools: Thus might have missed it out... I have added the tools for EDA. Thank you.